climada/hazard/tc_tracks.py

```diff
@@ -2089,6 +2088,292 @@ def _one_interp_data(track, time_step_h, land_geom=None):
+        self,
+        res: int = 5,
+        bounds: tuple = None,
+        genesis: bool = False,
+        filter_tracks: bool = True,
+        wind_min: float = None,
+        wind_max: float = None,
+    ) -> tuple[np.ndarray, np.ndarray, np.ndarray]:
+        """Compute tropical cyclone track density. Before using this function,
+        apply the same temporal resolution to all tracks by calling :py:meth:`equal_timestep` on
+        the TCTrack object. Due to the computational cost of the this function, it is not
+        recommended to use a grid resolution higher tha 0.1°. First, this function creates 2D bins
+        of the specified resolution (e.g. 1° x 1°). Second, since tracks are not lines but a series
+        of points, it counts the number of points per bin. Lastly, it returns the absolute count
+        per bin. To plot the output of this function use :py:meth:`plot_track_density`.
+
+        Parameters:
+        ----------
+        tc_track: TCTracks object
+            track object containing a list of all tracks
+        res: int (optional), default: 5°
+            resolution in degrees of the grid bins in which the density will be computed
+        bounds: tuple, (optional) dafault: None
+            (lon_min, lat_min, lon_max, lat_max) latitude and longitude bounds.
+        genesis: bool, (optional) default = False
+            If true the function computes the track density of only the genesis location of tracks
+        filter_tracks: bool (optional) default: True
+            If True the track density is computed as the number of different tracks crossing a grid
+            cell. If False, the track density takes into account how long the track stayed in each
+            grid cell. Hence slower tracks increase the density if the parameter is set to False.
+        wind_min: float (optional), default: None
+            Minimum wind speed above which to select tracks (inclusive).
+        wind_max: float (optional), default: None
+            Maximal wind speed below which to select tracks (exclusive if wind_min is also provided,
+            otherwise inclusive).
+        Returns:
+        -------
+        hist_count: np.ndarray
+            2D matrix containing the the absolute count per grid cell of track point or the
+            normalized number of track points, depending on the norm parameter.
+        lat_bins: np.ndarray
+            latitude bins in which the point were counted
+        lon_bins: np.ndarray
+            longitude bins in which the point were counted
+
+        Example:
+        --------
+        >>> tc_tracks = TCTrack.from_ibtracs_netcdf("path_to_file")
+        >>> tc_tracks.equal_timestep(time_step_h = 1)
+        >>> hist_count, _, _ = compute_track_density(tc_track = tc_tracks, res = 1)
+        >>> ax = plot_track_density(hist_count)
+
+        """
+
+        limit_ratio: float = (
+            1.12 * 1.1
+        )  # record tc speed 112km/h -> 1.12°/h + 10% margin
```

Comment:
I guess having this limit hard coded is fine here. But out of curiosity: why 112km/h / 1.12 degree? is this about the expected value in tropical cyclones?

Reply:
The fastest recorded TC to my knowledge was traveling at 112km/h. If the time step is 1h we want the grid cell to be at least 112km x 112km, so that the points in a track do not skip a grid cell. Which means there is a ratio between max speed, time step and resolution that should not be crossed (I added 10% to the max speed as margin). Here I assume no TC will ever be faster than 112km/h + 10% margin.

Around the equator 112km equates to roughly 1°, so 112km/h becomes 1°/h .

(I should have written 1° instead of 1.12°)